store/view_utilities.py

```diff
@@ -0,0 +1,50 @@
+from collections import defaultdict
+from store.models import Order, OrderItem, Product, Store
+
+
+def get_products_and_quantities_from_bag(request):
+    products = []
+    if "bag" in request.session:
+        for product in request.session["bag"]:
+            product_instance = Product.objects.get(id=product["product_id"])
+            products.append(
+                {
+                    "product": product_instance,
+                    "quantity": product["quantity"],
+                    "image": product_instance.image.url,
+                }
+            )
+        return products
+    else:
+        return products
+    
+def get_order_items_by_store(products_in_bag):
```

Comment:
The function is well-structured and leverages defaultdict effectively. No major changes needed